### ISSUE TITLE
Merge pending processed region list

### ISSUE THREAD

**Issue by rleungx** [tikv/pd#8378]
## Enhancement Task
We have many extra lists to store the regions that we want to check with the higher priority in every round of patroling region. It's duplicated and we need to unify them.

**Comment by okJiang** [comment]
Is this issue completed?

**Comment by rleungx** [comment]
No, I plan to merge it to the priority queue, but the current one is not implemented correctly.
